authentication_service/middleware/middleware.go

```diff
@@ -0,0 +1,39 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	helper "github.com/leechongyan/Studtor_backend/authentication_service/helpers/account"
+
+	"github.com/gin-gonic/gin"
+)
+
+func Authentication() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		clientToken := c.Request.Header.Get("Authorization")
+		if clientToken == "" {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": fmt.Sprintf("No Authorization Header provided")})
+			c.Abort()
+			return
+		}
+
+		splitToken := strings.Split(clientToken, "Bearer ")
```

Comment:
the golang docs says that 
> If s does not contain sep and sep is not empty, Split returns a slice of length 1 whose only element is s.

because this is the first step in our main routes, there exists the possibility of `splitToken` being of length 1. Would this then crash iwth an indexing error?

Reply:
ok will change